### DISCUSSION TITLE
Meta: Some interesting pointers to borrow ideas

### DISCUSSION THREAD

**Discussion by sirinath** [uwu-tech/Kind#231] [General]
- https://groupoid.space/
- https://n2o.dev/
- https://erp.uno/
- https://o83.github.io/platform.rs/
- https://www.microsoft.com/en-us/research/project/bosque-programming-language/
- https://dafny-lang.github.io/dafny/
- https://koka-lang.github.io/koka/doc/index.html
- https://www.unisonweb.org/
- http://why3.lri.fr/
- https://www.fstar-lang.org/
- https://github.com/microsoft/verona
- https://www.luna-lang.org/
- http://www.ats-lang.org/

**Comment by rigille** [comment]
among these what i like the most are Koka's memory management, and F*'s automation

**Reply by dumblob** [reply]
Koka's memory management seems similar (but still less optimized) to [Lobster's memory management](http://strlen.com/lobster/ ) and to [Nim's ARC](https://nim-lang.org/blog/2020/10/15/introduction-to-arc-orc-in-nim.html ). Somewhat similar approach is also ["Vaporization"](https://slightknack.dev/blog/vaporization/ ).
